src/layouts/home-assistant-main.html

```diff
@@ -87,83 +88,89 @@
 
 <script>
 (function () {
-  var NON_SWIPABLE_PANELS = ['kiosk', 'map'];
-  Polymer({
-    is: 'home-assistant-main',
-
-    properties: {
-      hass: Object,
-
-      narrow: Boolean,
-
-      route: {
-        type: Object,
-        observer: '_routeChanged',
-      },
-      routeData: Object,
-      routeTail: Object,
-      statesRouteTail: Object,
-
-      dockedSidebar: {
-        type: Boolean,
-        computed: 'computeDockedSidebar(hass)',
-      },
-    },
-
-    listeners: {
-      'hass-open-menu': 'handleOpenMenu',
-      'hass-close-menu': 'handleCloseMenu',
-      'hass-start-voice': 'handleStartVoice',
-    },
-
-    _routeChanged: function () {
+  const NON_SWIPABLE_PANELS = ['kiosk', 'map'];
+
+  class HomeAssistantMain extends window.hassMixins.EventsMixin(Polymer.Element) {
+    static get is() { return 'home-assistant-main'; }
+
+    static get properties() {
+      return {
+        hass: Object,
+
+        narrow: Boolean,
+
+        route: {
+          type: Object,
+          observer: '_routeChanged',
+        },
+        routeData: Object,
+        routeTail: Object,
+        statesRouteTail: Object,
+
+        dockedSidebar: {
+          type: Boolean,
+          computed: 'computeDockedSidebar(hass)',
+        },
+      };
+    }
+
+    ready() {
+      super.ready();
+      this.addEventListener('hass-open-menu', this.handleOpenMenu);
```

Comment:
Is `this` auto-bound to those callbacks?

Reply:
Good question. They seem to be working in the browser, although I'm trying to find some documentation that supports this is how ES6 classes work.

Reply:
I'm just going to add arrow functions to be safe.

Reply:
Here: https://stackoverflow.com/questions/36489579/this-within-es6-class-method
It says that `this` of event listeners is the object that got the event, which in this case is the same `home-assistant`